segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/operations/StreamSegmentAppendOperation.java

```diff
@@ -39,6 +43,12 @@ public class StreamSegmentAppendOperation extends StorageOperation implements At
     protected BufferView data;
     protected AttributeUpdateCollection attributeUpdates;
 
+    // Hash of the Append contents. Note that this is only used internally within the Segment Store to
+    //check data integrity and should not be considered for serialization.
+    @Getter
+    @Setter
+    private transient long contentHash = AppendIntegrityChecker.NO_HASH;
```

Comment:
This isn't effective. You aren't storing it in Tier 1. So anything that gets written to LTS after a recovery will not have this set.

I also think you meant `volatile` and not `transient`. The latter has no effect in our serialization.

Reply:
The goal of this (part) of the PR is to make sure that what we get at the beginning of the ingestion pipeline (e.g., `SegmentContainer` level) is the same data as the one we write to LTS. This is the whole purpose of this data integrity check. This is why we i) calculate the hash when an append is first ingested, ii) propagate that hash internally attaching it to the Append object itself (without actually writing the hash to Bookkeeper), and iii) validate the hashes at the `SegmentAggregator` level comparing the hashes of the Appends contained on a chuck of data read from cache to be written to LTS.

You are right in that, in the case of container recovery, this is not validating the integrity of Appends recovered from Bookkeeper (we are not impacting compatibility neither). But in the normal case, this is supposed to validate that data is not corrupted internally in Pravega due to a bug or something like that. Maybe we can think about how to calculate the hashes during container recovery, so all the cases would be covered, right?

I actually mean `transient`, because I wanted to make it very clear that this field should not be serialized as part of the Append. I see that the serialization is done at the bottom of this class, but I wanted to make it explicit, so nobody in the future tries to serialize this, as it is not intended to be serialized.

Reply:
Note that we also compute hashes of data read from Bookkeeper (not to mention that Bookkeeper itself has checksums to validate integrity of data on its own).